### PR TITLE
Transform readonly-combobox into select-listbox

### PR DESCRIPTION
**Pull request by yomed**
**WIP** -- Opening a PR to see if this is the direction we want to go.

## Description

- copy readonly-combobox styles to select-listbox
- clean up and share code with the native select where possible
- update docs

## Context

This gives a general idea of what the merge will look like, where the code is shared with native select rather than with readonly combobox. We can decide if we want to move forward on this refactor, or do something else.

### CODE REVIEW

**Comment by msendlakowski** [comment]
@yomed  Yeah.  That's what I was thinking.  And when we get to ebayUI we can try to make one component that obfuscates the differences between menu/select/select-listbox/combobox for the applications. 

Is there a reason the menu uses a `<button class="expand-btn">` while you used an `<input type="button">` for this?  Seems like something that could be unified.

File: src/less/select/ds6/select.less [hunk 1]
```diff
@@ -31,11 +21,8 @@ span.select__icon {
     }
 }
 
-[dir="rtl"] {
-    .select& > select {
-        padding-left: 30px;
-        padding-right: 16px;
-    }
+.select[dir="rtl"] > select {
+    padding: 0 16px 0 32px;
```
Comment (seangates) [inline src/less/select/ds6/select.less]: Why did the padding on the left change to `32px` from `40px`?

File: src/less/select/ds4/select.less [hunk 2]
```diff
@@ -65,14 +72,11 @@ span.select__icon {
     border-color: @ds4-color-core-blue;
 }
 
-[dir="rtl"] {
-    .select& > select {
-        padding-left: 40px;
-        padding-right: 16px;
-    }
+.select[dir="rtl"] > select {
+    padding: 0 16px 0 36px;
```
Comment (seangates) [inline src/less/select/ds4/select.less]: Why did the padding on the left change to `32px` from `40px`?

File: src/less/select/ds4/select.less [hunk 3]
```diff
@@ -1,25 +1,16 @@
...
-
 .select > select {
     background-color: @ds4-color-core-white;
     border-color: @ds4-color-core-gray-silver;
     border-radius: 3px;
     color: @ds4-color-core-gray-davys;
-    font-family: @ds4-font-family-base;
-    padding: 0 40px 0 16px;
+    font-size: @ds4-font-size-14;
+    padding: 0 36px 0 16px;
```
Comment (seangates) [inline src/less/select/ds4/select.less]: Why did the padding on the left change to `36px` from `40px`?

File: src/less/select/ds4/select.less [hunk 4]
```diff
@@ -37,26 +28,42 @@ span.select__icon {
...
 }
 
-.select--small {
-    font-size: @ds4-font-size-12;
+.select__control > input {
+    border-color: @ds4-color-core-gray-silver;
+    border-radius: 3px;
+    font-size: @ds4-font-size-14;
+    height: @ds4-height-control-medium;
+    padding: 0 36px 0 16px;
```
Comment (seangates) [inline src/less/select/ds4/select.less]: Why did the padding on the right change to `36px` from `40px`?

Comment (seangates) [reply src/less/select/ds4/select.less]: Oh, this wasn't a change, but it also has the `36px` instead of `40px` which has been there throughout.

**Comment by ianmcburnie** [comment]
The direction I want to go is:

* revert recent combobox changes (I know the readonly textfield was causing problems in iOS, but we need to go back to input type=text for an editable combobox)
* create a new listbox module that uses a button (essentially mirroring the recent combobox changes). We will tell teams to migrate from ebay-combobox to ebay-listbox.

This keeps us consistent with W3C patterns (I will be updating the MIND patterns to reflect this too). Of course the downside is that we potentially introduce yet more confusion with another type of dropdown on the Skin side of things. Hopefully on the eBayUI side of things we can introduce a new component that can help abstract the underlying differences.

**Comment by ianmcburnie** [comment]
@yomed I think what you have matches pretty closely with what I'm saying, just that I want to call it `listbox` (to match W3C authoring patterns) rather than `select`...?

**Comment by yomed** [comment]
Yeah, the idea was to unify `select` and `select`-like in one component.

**Comment by ianmcburnie** [comment]
I've created a new branch from this one, called `396-listbox`. I will continue the work in there and close this PR.

**Pull request closed without merging**